src/components/storage-plan-modal.js

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types';
+import ByteInput from './byte-input.js';
+import { Form, Input, Modal } from 'antd';
+import { useTranslation } from 'react-i18next';
+import React, { useEffect, useState } from 'react';
+import { baseStoragePlanShape } from '../ui/default-prop-types.js';
+
+const FormItem = Form.Item;
+
+function StoragePlanModal({ isVisible, storagePlan, storagePlanNamesInUse, onOk, onCancel }) {
+  const [form] = Form.useForm();
+  const { t } = useTranslation('storagePlanModal');
+  const [loading, setLoading] = useState(false);
+
+  const nameValidationRules = [
+    {
+      required: true,
+      message: t('nameRequired'),
```

Comment:
`nameRequired`, also known as "translation required" 😄

Reply:
That's so true :laughing: Added!